src/lib/proxy_task.rs

```diff
@@ -0,0 +1,57 @@
+use std::env;
+
+use crate::{logger, profile, types::Task};
+
+pub(crate) fn create_proxy_task(
```

Comment:
you should also move all create_proxy tests from runner_test to new proxy_task_test file

Reply:
Yes, of course haha I knew there was _something_. All fixed.